### PR TITLE
feat(event): #3 - added base structures for CoreEvent and other essential data structures

### PR DESCRIPTION
**Pull request by abtinokhovat**

### CODE REVIEW

**Review by gohossein — COMMENTED** [review]
Please change `scope` in Pull-Request title.
```
feat(core): added...
```
or
```
feat(event): added..
```

File: .gitignore [hunk 1]
```diff
@@ -20,4 +20,6 @@
 # Go workspace file
 go.work
 
-.idea
\ No newline at end of file
+.idea
+
+go.sum
```
Comment (gohossein) [inline .gitignore:25]: I'm not sure we should include `go.sum` in `.gitignore` file.

**Review by gohossein — COMMENTED** [review]
Great start!

**Comment by abtinokhovat** [comment]
> Please change `scope` in Pull-Request title.
> 
> ```
> feat(core): added...
> ```
> 
> or
> 
> ```
> feat(event): added..
> ```

should I add the issue number in the description?

File: .gitignore (hunk 1 above)
Comment (abtinokhovat) [reply .gitignore:25]: isn't this file generated in each dev environment? like package_lock.json for npm package manager?

Comment (gohossein) [reply .gitignore:25]: No, this isn't lock file.
https://github.com/golang/go/wiki/Modules#releasing-modules-all-versions

**Comment by gohossein** [comment]
> > Please change `scope` in Pull-Request title.
> > ```
> > feat(core): added...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or
> > ```
> > feat(event): added..
> > ```
> 
> should I add the issue number in the description?

Good question, I guess there is no way to directly link issue to pull-request, so I guess adding the name and link of issue in pull request description is good idea.
Actually you have access to organization and would be great if you can create a pull-request template to use for all PRs.

**Comment by abtinokhovat** [comment]
> > > Please change `scope` in Pull-Request title.
> > > ```
> > > feat(core): added...
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > or
> > > ```
> > > feat(event): added..
> > > ```
> > 
> > 
> > should I add the issue number in the description?
> 
> Good question, I guess there is no way to directly link issue to pull-request, so I guess adding the name and link of issue in pull request description is good idea. Actually you have access to organization and would be great if you can create a pull-request template to use for all PRs.

adding issue number in PR description or link will add a hyperlink to the issue, in more simple terms we can use `#` to tag issues or PRs

File: .gitignore (hunk 1 above)
Comment (abtinokhovat) [reply .gitignore]: Done,  thanks for the documentation.
I deleted the invalid commit.

**Comment by gohossein** [comment]
Please squash your commits

**Comment by gohossein** [comment]
Are you sure adding issue number to PR title (#3 here)  will link it to the issue?

**Comment by abtinokhovat** [comment]
> Are you sure adding issue number to PR title (#3 here) will link it to the issue?

Yep, if you are watching this on github it will also shows you the issue when you hover on it

CleanShot 1402-09-02 at 23 39 09

**Review by gohossein — APPROVED** [review]
Great

**Pull request merged**